src/main/java/org/broadinstitute/consent/http/service/TDRService.java

```diff
@@ -33,4 +38,18 @@ public ApprovedUsers getApprovedUsersForDataset(Dataset dataset) {
 
         return new ApprovedUsers(approvedUsers);
     }
+
+    public List<Integer> getDatasetIdsByIdentifier(List<String> identifiers) {
+        List<Integer> datasetIds = identifiers
+                .stream()
+                .filter(identifier -> !identifier.isBlank())
+                .map(identifier -> datasetDAO.findDatasetByAlias(Dataset.parseIdentifierToAlias(identifier)))
```

Comment:
We want to avoid making individual DAO calls within a map. If a user provides, let say 30, identifiers, then we'd be making 30 individual calls to the database.

What should happen is that you need to make a new DAO method that takes an ArrayList of identifiers as a parameter and perform a bulk fetch. This way, regardless of the number of identifiers, you'd only be making a single database call every time.

Look at `findVotesByElectionIds` in `VoteDAO.java` for an example on how to write up these types of DAO methods.

Reply:
Added a new DatasetDAO method that takes in a list of identifiers and returns a list of Datasets.